postgres/cloudpostgres/cloudpostgres.go

```diff
@@ -55,6 +61,49 @@ type Params struct {
 	TraceOpts []ocsql.TraceOption
 }
 
+// URLOpener opens GCP PostgreSQL URLs
+// like "cloudpostgres://user:password@myproject.us-central1.instanceId/mydb".
+type URLOpener struct {
+	// TraceOpts contains options for OpenCensus.
+	TraceOpts []ocsql.TraceOption
+}
+
+// Scheme is the URL scheme cloudpostgres registers its URLOpener under on
+// postgres.DefaultMux.
+const Scheme = "cloudpostgres"
+
+func init() {
+	postgres.DefaultURLMux().RegisterPostgres(Scheme, &URLOpener{})
+}
+
+// OpenPostgresURL opens a new GCP database connection wrapped with OpenCensus instrumentation.
+func (uo *URLOpener) OpenPostgresURL(ctx context.Context, u *url.URL) (*sql.DB, error) {
+	vals := u.Query()
+	u.RawQuery = vals.Encode()
+	creds, err := gcp.DefaultCredentials(ctx)
+	client, err := gcp.NewHTTPClient(http.DefaultTransport, creds.TokenSource)
+	certSource := cloudsql.NewCertSource(client)
+	gcpInstance := strings.Split(u.Host, ".")
+	if len(gcpInstance) < 3 {
```

Comment:
Having more components seems bad too.

```suggestion
	if len(gcpInstance) != 3 {
```